### PR TITLE
Robo HAT MM1 updates v1.8 board

### PR DESCRIPTION
**Pull request by wallarug**
In this commit:
 - new pin layout for Robo HAT
 - two boards included (SAMD21 and SAMD51)
 - added build linker file for SAMD51 with external flash + crystalless
 - updated names to reflect m0 and m4 chips
 - updated travis.yml

Should not impact on other boards AFAIK.

### CODE REVIEW

**Comment by wallarug** [comment]
Hi there!

I just made a few small changes to pins.c.  These should not conflict with existing boards.

Please let me know when this is likely to be merged in

😄   Thanks!

File: ports/atmel-samd/boards/robohatmm1_m4/mpconfigboard.mk [hunk 1]
```diff
@@ -0,0 +1,31 @@
+LD_FILE = boards/samd51x19-bootloader-external-flash-crystalless.ld
```
Comment (tannewt) [inline ports/atmel-samd/boards/robohatmm1_m4/mpconfigboard.mk]: I don't think you need a new linker file so please reuse the existing one.

We run most M4s crystal-less now I think. The crystalless linker script is only needed with CALIBRATE_CRYSTALLESS on M0 which stores crystal calibration into flash.

Comment (wallarug) [reply ports/atmel-samd/boards/robohatmm1_m4/mpconfigboard.mk]: No problem.  I would want to test this first though.
```
LD_FILE = boards/samd51x19-bootloader-external-flash.ld
```

Comment (wallarug) [reply ports/atmel-samd/boards/robohatmm1_m4/mpconfigboard.mk]: Tested - Done.  Going to remove the extra linker script

**Comment by wallarug** [comment]
@tannewt  requesting review 😎

**Pull request merged**